sonar-db/src/main/java/org/sonar/db/measure/PastMeasureDto.java

```diff
@@ -19,28 +19,23 @@
  */
 package org.sonar.db.measure;
 
-import java.util.Objects;
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
 
+import static java.util.Objects.requireNonNull;
+
 public class PastMeasureDto {
 
-  private Long id;
-  private Double value;
-  private Integer metricId;
-  private Integer personId;
+  private int metricId;
 
-  public Long getId() {
-    return id;
-  }
+  @CheckForNull
+  private Double value;
 
-  public PastMeasureDto setId(Long id) {
-    this.id = id;
-    return this;
-  }
+  @CheckForNull
+  private Long personId;
 
   public double getValue() {
-    Objects.requireNonNull(value);
+    requireNonNull(value);
```

Comment:
if value can't effectively be null in DTO, you could replace type `Double` for field value with type `double` with a default value, there would clearly be memory gain